libs/dh/shared/util-apollo/src/lib/query.ts

```diff
@@ -200,7 +246,8 @@ export function query<TResult, TVariables extends OperationVariables>(
         .pipe(
           map(({ data }) => data),
           exists(), // The data should generally be available, but types says otherwise
-          withLatestFrom(refReplay$), // Ensure the ref (and cache) is ready,
+          withLatestFrom(refReplay$), // Ensure the ref (and cache) is ready
+          delayWhen(() => result()), // Do not attempt to merge when result is loading
           takeUntilDestroyed(destroyRef), // Stop the subscription when the component is destroyed
           takeUntil(reset$) // Or when resetting the query
         )
```

Comment:
praise: good that we have tests to cover these functions. Otherwise it's untouchable 😝

Reply:
Not sure if sarcastic or not, but now there are actual tests for this specific scenario 😆 
https://github.com/Energinet-DataHub/greenforce-frontend/pull/3617/commits/2d62112e3ca1dfbf39dcc186d529be67522267f5

Reply:
Wasn't sarcastic, sorry for making it ambiguous. It was a genuine comment about the tests that come along with the functions. I think they're invaluable to have, and make future changes easier. So great work again. 👍